solidity/test/integrator/TBTCDepositorProxy.test.ts

```diff
@@ -0,0 +1,317 @@
+            await createSnapshot()
+
+            await bridge.sweepDeposit(fixture.expectedDepositKey)
+
+            tx = await depositorProxy.finalizeDepositPublic(
+              fixture.expectedDepositKey
+            )
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should remove the deposit from pending", async () => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+            expect(
+              await depositorProxy.pendingDeposits(fixture.expectedDepositKey)
+            ).to.be.false
+          })
+
+          it("should emit the DepositFinalized event", async () => {
+            await expect(tx)
+              .to.emit(depositorProxy, "DepositFinalized")
+              .withArgs(
+                fixture.expectedDepositKey,
+                expectedTbtcAmount,
+                await lastBlockTime()
+              )
+          })
+
+          it("should call onDepositFinalized with proper params", async () => {
+            await expect(tx)
+              .to.emit(depositorProxy, "OnDepositFinalizedCalled")
+              .withArgs(
+                fixture.expectedDepositKey,
+                expectedTbtcAmount,
+                fixture.extraData
+              )
+          })
+        })
+
+        context("when the deposit is optimistically minted", () => {
+          let tx: ContractTransaction
+
+          before(async () => {
+            await createSnapshot()
+
+            await tbtcVault.createOptimisticMintingRequest(
+              fixture.expectedDepositKey
+            )
+
+            await tbtcVault.finalizeOptimisticMintingRequest(
+              fixture.expectedDepositKey
+            )
+
+            tx = await depositorProxy.finalizeDepositPublic(
+              fixture.expectedDepositKey
+            )
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should remove the deposit from pending", async () => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+            expect(
+              await depositorProxy.pendingDeposits(fixture.expectedDepositKey)
+            ).to.be.false
+          })
+
+          it("should emit the DepositFinalized event", async () => {
+            await expect(tx)
+              .to.emit(depositorProxy, "DepositFinalized")
+              .withArgs(
+                fixture.expectedDepositKey,
+                expectedTbtcAmount,
+                await lastBlockTime()
+              )
+          })
+
+          it("should call onDepositFinalized with proper params", async () => {
+            await expect(tx)
+              .to.emit(depositorProxy, "OnDepositFinalizedCalled")
+              .withArgs(
+                fixture.expectedDepositKey,
+                expectedTbtcAmount,
+                fixture.extraData
+              )
+          })
+        })
+      })
+    })
+  })
```

Comment:
Would you be interested in adding tests for `calculateDepositKey` function to cover a scenario where the fees are set to zero?

Reply:
Yeah, it's good to cover those paths just in case. See 9efd0d0c